test/expected/insert.out

```diff
@@ -755,3 +755,46 @@ SELECT 1 \g | grep -v "Planning" | grep -v "Execution"
                        ->  Result (actual rows=1 loops=1)
 (8 rows)
 
+-- test ON CONFLICT with prepared statements and functions
```

Comment:
I think this belongs in upsert.sql. Also It's better to have both excluded.X plus table.X in the upsert clause (you'll see a lot of examples in upsert.sql). Finally, I'd like to see the resulting table rows via a select (we've seen weird behavior before that's caught by simple selects).